### ISSUE TITLE
Security Concerns and Questions

### ISSUE THREAD

**Issue by kiranranjane** [moosefs/moosefs#285]
Hello Team,

I have a few questions related to security and I hope you guys can help me out.

1) I could not find anything in docs on how to enable encryption on rest and while data is in transit between nodes. Is there a way to do it using external tools if Moosefs does not have this feature?

2) Any eta if this feature will be added to Moosefs?

3) Also, is there a way we can add a passphrase to mount points using mfsmount or any other tool because if someone is able to access the server then whole storage is exposed and could be a security breach at the storage level, We want to restrict the access of the mountpoints if that is possible.

Regards
Kiran

**Comment by acid-maker** [comment]
1. Encryption in MFS is not implemented (yet)
2. No ETA - this feature is of course on our roadmap but with rather low priority
3. Yes. In 'mfsexports.cfg' you can add passphrase (see man mfsexports.cfg and mfsexports.cfg.sample file)
4. It is also good idea to add passphrase for your chunkservers (define AUTH_CODE in your mfsmaster.cfg and mfschunkserver.cfg on each chunkserver)

**Comment by tanero** [comment]
Is there  any solution for protecting the access between master and metalogger, like chunk and master by using AUTH_CODE ?

**Comment by chogata** [comment]
No, there is no auth code for metaloggers, but it's a good idea to add one, we're adding it to the feature request list. Meanwhile, you can use an auxiliary solution, a simple firewall can make sure that only specified connection between two specific IPs on a specified port can happen.
